File: Source/MessageType.m
//
//  MessageType.m
//  RongCloudIMManagerExample
//
//  Created by Mac on 2017/9/4.
//  Copyright © 2017年 LiYou. All rights reserved.
//

#import "MessageType.h"

@implementation MessageType

- (NSData *)encode {
    
    NSMutableDictionary *dataDict = [NSMutableDictionary dictionaryWithCapacity:20];
    
    [dataDict setValue:self.house_name forKey:@"house_name"];
    [dataDict setValue:@(self.house_id) forKey:@"house_id"];
    [dataDict setValue:@(self.city_id) forKey:@"city_id"];
    [dataDict setValue:@(self.sprice) forKey:@"sprice"];
    [dataDict setValue:@(self.sale_count) forKey:@"sale_count"];
    [dataDict setValue:self.isfirst forKey:@"isfirst"];
    [dataDict setValue:self.extra forKey:@"extra"];
    
    return [NSJSONSerialization dataWithJSONObject:dataDict options:NSJSONWritingPrettyPrinted error:nil];
}

- (void)decodeWithData:(NSData *)data {
    
    NSDictionary *dataDict = [NSJSONSerialization JSONObjectWithData:data options:NSJSONReadingMutableLeaves error:nil];
    self.house_name = dataDict[@"house_name"];
    self.house_id   = [dataDict[@"house_id"] integerValue];
    self.city_id    = [dataDict[@"city_id"] integerValue];
    self.sprice     = [dataDict[@"sprice"] integerValue];
    self.sale_count = [dataDict[@"sale_count"] integerValue];
    self.isfirst    = dataDict[@"isfirst"];
    self.extra      = dataDict[@"extra"];
    
}

- (NSString *)conversationDigest {
    return self.house_name;
}

@end

File: Source/MessageType.swift
//
//  MessageType.swift
//  rabbit
//
//  Created by PixelShi on 2017/1/16.
//  Copyright © 2017年 lychinahouse. All rights reserved.
//

import UIKit

class MessageType: RCMessageContent {
    var house_name: String = ""
    var house_id: Int = 0
    var city_id: Int = 0
    var sprice: Int = 0
    var sale_count: Int = 0
    var isfirst: String = ""
    var extra: String = ""

    override func encode() -> Data {
        let dataDict: NSMutableDictionary = NSMutableDictionary()

        dataDict.setValue(house_name, forKey: "house_name")
        dataDict.setValue(house_id, forKey: "house_id")
        dataDict.setValue(city_id, forKey: "city_id")
        dataDict.setValue(sprice, forKey: "sprice")
        dataDict.setValue(sale_count, forKey: "sale_count")
        dataDict.setValue(isfirst, forKey: "isfirst")
        dataDict.setValue(extra, forKey: "extra")
        do {
            let data = try JSONSerialization.data(withJSONObject: dataDict, options: JSONSerialization.WritingOptions.init(rawValue: 1))
            return data
        } catch {

        }
        return Data()
    }

    override func decode(with data: Data) {
        do {
            let dic: [String: Any?] = try JSONSerialization.jsonObject(with: data, options: JSONSerialization.ReadingOptions.init(rawValue: 1)) as! [String: Any?]

            if let id = dic["house_name"] as? String {
                house_name = id
            }
            if let id = dic["house_id"] as? Int {
                house_id = id
            }
            if let id = dic["city_id"] as? Int {
                city_id = id
            }
            if let id = dic["sprice"] as? Int {
                sprice = id
            }
            if let id = dic["sale_count"] as? Int {
                sale_count = id
            }
            if let id = dic["isfirst"] as? String {
                isfirst = id
            }
            if let id = dic["extra"] as? String {
                extra = id
            }
        } catch {

        }
    }

    override func conversationDigest() -> String! {
        return self.house_name
    }

    override class func getObjectName() -> String! {
        return "2boss:house"
    }

}
